File: bubble_sort/bubble_sort.py
# Python program for implementation of Bubble Sort, in a ascending or descending way

def bubble_sort(array, cond):

    # Traverse through all array elements
    for i in range(len(array)):
        swapped = False

        if cond == 'asc':
            # Comparison with the next element
            for j in range(len(array)-1):
                if array[j] > array[j+1]:
                    array[j], array[j+1] = array[j+1], array[j]
                    swapped = True
        if cond == 'desc':
            # Comparison with the next element
            for j in range(len(array)-1):
                if array[j] < array[j+1]:
                    array[j], array[j+1] = array[j+1], array[j]
                    swapped = True
        if swapped == False:
            break

            


if __name__ == '__main__':
    # Code to test the Bubble sort
    length = int(input("Please, input the number of elements in the array: "))
    cond = input("Please, input the condition of sorting ['asc', 'desc']: ")

    array = []
    for i in range(length):
        element = int(input(f"Element {i+1} in the array: "))
        array.append(element)

    print("")
    print("Unsortered Array")
    [print(i, end=" ") for i in array]
    print("\n")
    print("Sortered Array")
    bubble_sort(array, cond)
    [print(i, end=" ") for i in array]

File: bubble_sort/bubble_sort.c
#include <stdio.h>

void bubble_sort(int array[], int len){
    //Auxiliary Variable
    int next = 0;

    // Traverse through all array elements
    for (int i = 0; i < len; i++){
        //Work with the next elements
        for (int j = 0; j < (len-1);j++){
            // The swap takes place
            if (array[j] > array[j+1]){
                next = array[j];
                array[j] = array[j+1];
                array[j+1] = next;
            }
        }
    }
}

int main()
{
    //disordered array
    int array[6] = {8,3,1,42,12,5};

    // Ordenation function utilizing 'bubble sort'
    bubble_sort(array, 6);

    // Presentation of the ordered vector
    for (int i = 0; i < 6; i++){
        printf("%d\n", array[i]);
    }
    return 0;
}